### PR TITLE
check ->argsz for VFIO_USER_DIRTY_PAGES

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: lib/libvfio-user.c [hunk 1]
```diff
@@ -894,18 +894,19 @@ handle_dirty_pages_get(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
...
                 msg->in_size, dirty_pages_in->argsz);
         return ERROR_INT(EINVAL);
     }
 
     range_in = msg->in_data + sizeof(*dirty_pages_in);
 
-    /* NB: this is bound by MAX_DMA_SIZE. */
-    argsz = sizeof(*dirty_pages_out) + sizeof(*range_out) +
-            range_in->bitmap.size;
+    argsz = satadd_u64(sizeof(*dirty_pages_out) + sizeof(*range_out),
```
Comment (tmakatos) [inline lib/libvfio-user.c:907]: So if there's overflow we'll silently truncate `argsz`? Shouldn't we just fail?

Comment (jlevon) [reply lib/libvfio-user.c:907]: we'll fail later on when the dma controller checks bitmap.size but we don't have that info here. So we just need something safe against overflow.

Comment (tmakatos) [reply lib/libvfio-user.c:907]: Is this what the `MAX_DMA_SIZE` comment is about? It's a bit unclear.

Comment (jlevon) [reply lib/libvfio-user.c:907]: It was, but that comment was misleading. Let me fix

**Review by tmakatos — APPROVED** [review]

**Pull request merged**